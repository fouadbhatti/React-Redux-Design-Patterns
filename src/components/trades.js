import React, { memo } from 'react';

export function Trades({ tradesList = [] }) {
  return (
        <table>
          <tbody>
            <tr>
              <th>Time</th>
              <th>Price</th>
              <th>Amount</th>
            </tr>
            {
              tradesList.map((item, index) =>
                <tr id={item.id} key={index}>
                  <td>
                    { new Date(item.mts).toLocaleTimeString() }
                  </td>
                  <td>
                    { item.price }
                  </td>
                  <td>
                    { item.amount }
                  </td>
                </tr>
              )
            }
          </tbody>
        </table>
  );
}

export default memo(Trades);