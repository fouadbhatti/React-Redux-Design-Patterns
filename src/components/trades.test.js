import React from 'react';
import renderer from 'react-test-renderer';
import { Trades } from './trades';

describe('Trades Component Test', () => {
  it('renders as expected', () => {
    const tradeList = [
      {id: 349248445, mts: 1554913866540, amount: 0.01054537, price: 5299.5},
      {id: 349248364, mts: 1554913848803, amount: 0.01628212, price: 5299.5},
      {id: 349248325, mts: 1554913825823, amount: 0.1, price: 5299.5},
      {id: 349248320, mts: 1554913822349, amount: 0.05239461, price: 5299.5},
      {id: 349248317, mts: 1554913819804, amount: 0.004, price: 5299.5},
      {id: 349248308, mts: 1554913816230, amount: 0.0176, price: 5299.4},
      {id: 349248305, mts: 1554913814996, amount: 0.01170507, price: 5299.4},
      {id: 349248291, mts: 1554913812072, amount: 0.02298214, price: 5299.4},
      {id: 349248280, mts: 1554913809511, amount: 0.052, price: 5298.2},
      {id: 349248281, mts: 1554913809511, amount: 0.095, price: 5298.3}
    ];
    const componentTree = renderer.create(
      <Trades
        tradesList={tradeList}
      />)
    .toJSON();

    expect(componentTree).toMatchSnapshot();
  });
});