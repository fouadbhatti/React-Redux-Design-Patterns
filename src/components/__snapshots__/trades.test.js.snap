// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trades Component Test renders as expected 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Time
      </th>
      <th>
        Price
      </th>
      <th>
        Amount
      </th>
    </tr>
    <tr
      id={349248445}
    >
      <td>
        9:31:06 PM
      </td>
      <td>
        5299.5
      </td>
      <td>
        0.01054537
      </td>
    </tr>
    <tr
      id={349248364}
    >
      <td>
        9:30:48 PM
      </td>
      <td>
        5299.5
      </td>
      <td>
        0.01628212
      </td>
    </tr>
    <tr
      id={349248325}
    >
      <td>
        9:30:25 PM
      </td>
      <td>
        5299.5
      </td>
      <td>
        0.1
      </td>
    </tr>
    <tr
      id={349248320}
    >
      <td>
        9:30:22 PM
      </td>
      <td>
        5299.5
      </td>
      <td>
        0.05239461
      </td>
    </tr>
    <tr
      id={349248317}
    >
      <td>
        9:30:19 PM
      </td>
      <td>
        5299.5
      </td>
      <td>
        0.004
      </td>
    </tr>
    <tr
      id={349248308}
    >
      <td>
        9:30:16 PM
      </td>
      <td>
        5299.4
      </td>
      <td>
        0.0176
      </td>
    </tr>
    <tr
      id={349248305}
    >
      <td>
        9:30:14 PM
      </td>
      <td>
        5299.4
      </td>
      <td>
        0.01170507
      </td>
    </tr>
    <tr
      id={349248291}
    >
      <td>
        9:30:12 PM
      </td>
      <td>
        5299.4
      </td>
      <td>
        0.02298214
      </td>
    </tr>
    <tr
      id={349248280}
    >
      <td>
        9:30:09 PM
      </td>
      <td>
        5298.2
      </td>
      <td>
        0.052
      </td>
    </tr>
    <tr
      id={349248281}
    >
      <td>
        9:30:09 PM
      </td>
      <td>
        5298.3
      </td>
      <td>
        0.095
      </td>
    </tr>
  </tbody>
</table>
`;
